### PR TITLE
manage repositories from API and UI

### PR DESCRIPTION
**Pull request by prydonius**
- adds API methods for creating and deleting repositories
- adds basic UI view for managing repositories

screen shot 2017-08-14 at 20 28 02
screen shot 2017-08-14 at 20 28 19

### CODE REVIEW

File: src/ui/src/app/repositories/repositories.component.scss [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    &__new {
+      margin-left: 1em;
+      margin-bottom: 1em;
+      text-align: center;
+    }
+
+  }
+  // CSS from https://codepen.io/AllThingsSmitty/pen/MyqmdM
+  table {
+    border: 1px solid #ccc;
```
Comment (Angelmmiguel) [inline src/ui/src/app/repositories/repositories.component.scss]: Please, use the colors defined in `theme.scss` or a variation of them (`lighten`, `darken`...)

File: src/ui/src/app/repositories/repositories.component.scss [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    border-collapse: collapse;
+    margin: 0;
+    padding: 0;
+    width: 100%;
+    table-layout: fixed;
+  }
+  table caption {
```
Comment (Angelmmiguel) [inline src/ui/src/app/repositories/repositories.component.scss]: I think we don't need this rule. Are we going to use captions?

File: src/ui/src/app/repositories/repositories.component.scss [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    font-size: 1.5em;
+    margin: .5em 0 .75em;
+  }
+  table tr {
```
Comment (Angelmmiguel) [inline src/ui/src/app/repositories/repositories.component.scss]: Enclose the properties using SCSS:

```
table {
  border: ...;
  ...
  
  tr {
    ...
  }
}
```

File: src/ui/src/app/repositories/repositories.component.scss [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    text-transform: uppercase;
+  }
+  @media screen and (max-width: 600px) {
+    table {
+      border: 0;
+    }
+    table caption {
+      font-size: 1.3em;
+    }
+    table thead {
```
Comment (Angelmmiguel) [inline src/ui/src/app/repositories/repositories.component.scss]: What's the purpose of this? I don't know why we can't just use`display: none;`

File: src/ui/src/app/repositories/repositories.component.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    private config: ConfigService,
+    private mdIconRegistry: MdIconRegistry,
+    private sanitizer: DomSanitizer,
+    private dialogsService: DialogsService,
+    private dialog: MdDialog,
+    public snackBar: MdSnackBar,
+  ) {}
+
+  ngOnInit() {
+    this.mdIconRegistry.addSvgIcon(
```
Comment (Angelmmiguel) [inline src/ui/src/app/repositories/repositories.component.ts]: Are we using the `search`, `close` and `menu` icons? I only see the `delete` icon in the template

File: src/ui/src/app/repository-new/repository-new.component.scss [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    button + button {
+      margin-left: 1em;
+    }
+  }
+
+  &__form {
+    &__errors {
+      margin: 1em 0;
+      color: #f00;
```
Comment (Angelmmiguel) [inline src/ui/src/app/repository-new/repository-new.component.scss]: Use colors from `theme.scss`

File: src/ui/src/app/shared/models/repo.ts [hunk 7]
```diff
@@ -5,7 +5,7 @@ export class Repo {
 }
 
 export class RepoAttributes {
-  name: string;
-  URL: string;
-  source: string;
+  name: string = "";
```
Comment (Angelmmiguel) [inline src/ui/src/app/shared/models/repo.ts]: I think we are using single quotes in the repo. Just consistency :)

**Comment by Angelmmiguel** [comment]
A part of the comments, how does look the project in a mobile browser with the new element in the navbar?

File: src/ui/src/app/repositories/repositories.component.scss (hunk 1 above)
Comment (prydonius) [inline src/ui/src/app/repositories/repositories.component.scss]: I might need help from this, I was trying to import theme.scss but was getting an error. I'll try again and ping you on Slack!

File: src/ui/src/app/repositories/repositories.component.scss (hunk 2 above)
Comment (prydonius) [inline src/ui/src/app/repositories/repositories.component.scss]: No probably not, I just blanket copied the CSS from the site, will clean this up a bit!

**Comment by prydonius** [comment]
screen shot 2017-08-15 at 10 41 58

@Angelmmiguel it overflows a bit at a certain point. I was considering just leaving it as is and have @kemcake or you clean it up after, but if you prefer to have this fixed beforehand I can have a try.

**Comment by Angelmmiguel** [comment]
The code LGTM now. However, we must solve the header issue. It's fine if you do it in a separate PR. Up to you :)

**Pull request merged**